src/components/Calculator.js

```diff
@@ -1,29 +1,53 @@
+/* eslint-disable react/destructuring-assignment */
 /* eslint-disable react/prefer-stateless-function */
 import React, { Component } from 'react';
+import calculate from './logic/calculate';
 
 export default class Calculator extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      total: '0',
+      next: '',
+      operation: '',
+    };
+
+    this.clickHandler = this.clickHandler.bind(this);
+  }
+
+  clickHandler(e) {
+    const buttonName = e.target.id;
+    const { total, next, operation } = calculate(this.state, buttonName);
+    if (next === null && total === null) {
+      this.setState({ next, total: '0', operation });
+    } else {
+      this.setState({ next, total, operation });
+    }
+  }
+
   render() {
     return (
       <div className="Calc-container">
-        <div className="display-result">0</div>
-        <div className="clear pad">AC</div>
-        <div className="sign pad">+/-</div>
-        <div className="modular pad">%</div>
-        <div className="div operator">/</div>
-        <div className="seven pad">7</div>
-        <div className="eight pad">8</div>
-        <div className="nine pad">9</div>
-        <div className="mult operator">x</div>
-        <div className="four pad">4</div>
-        <div className="five pad">5</div>
-        <div className="six pad">6</div>
-        <div className="sub operator">-</div>
-        <div className="one pad">1</div>
-        <div className="two pad">2</div>
-        <div className="three pad">3</div>
-        <div className="plus operator">+</div>
-        <div className="zero pad">0</div>
-        <div className="equal operator">=</div>
+        <div className="display-result">{(this.state.next) ? this.state.next : this.state.total}</div>
```

Comment:
- [OPTIONAL] Implementation of tasks at hand is good. Let's do one improvement to enhance the user experience.

  ### The issue
  When the `=` button is pressed twice this weird behavior:
  
   - The display becomes blank and reduces in height as evident below
   
   | Before | After clicking on the `=` button twice |
   | --- | --- |
   | ![image](https://user-images.githubusercontent.com/45002654/154475335-cfb224a0-4e31-4a54-91cc-4978d248be9b.png) | ![image](https://user-images.githubusercontent.com/45002654/154475383-9d12786d-ff71-442c-9786-fd0797b1e4f5.png) |
   
   ### Suggestion
   We could solve that by doing this:
   ```JavaScript
   <div className="display-result">{this.state.next || this.state.total || '0'}</div>
   ```

Reply:
Thank you very much @Lameck1. I appreciate your reviews and I am going to implement the suggested improvement.